Source/SimConnect/Source/SimConnectClient.cpp

```diff
@@ -195,6 +195,8 @@ bool SimConnectClient::DestroySimObject( SimConnect::Handle& Handle )
 	const auto SimObjectId = SimObject->second;
 	const auto Result = SimConnect_AIRemoveObject( SimConnectHandle, SimObjectId, Handle.GetId() );
 
+	CreatedSimObjects.erase( SimObject );
```

Comment:
Does it make more sense to have this *after* the `if` statement below?  Since presumably, if `Result != S_OK`, then the SimObject may not have actually been removed?

Reply:
I did consider that, but then if we failed to remove it this time what will cause it to succeed the next time? It's kinda beyond our module's control if it failed to be removed sim-side so there's little we can do by keeping the simobject reference around on our side.

Reply:
Right, I see what you're saying -- if it failed once, doing the same thing again is likely to still exhibit the same behavior (and if that does indeed continue failing, there isn't much sense to let people keep trying to do that).

I suppose I was mostly thinking of the semantics of having something called `CreatedSimObjects` -- to me it makes sense to have that reflect the SimObjects that are currently still created in the sim.  However, I suppose if we really wanted that to reflect the state properly, we might want to use `SimConnect_SubscribeToSystemEvent` with the `ObjectRemoved` event (and also listen for `SIMCONNECT_RECV_EXCEPTION` failures as well)?

Anyways, for the purposes of working towards the first playable, it's definitely workable as-is -- just wanted to get some clarification. :)